pyaedt/filtersolutions_core/attributes.py

```diff
@@ -0,0 +1,1528 @@
+    """Selects type of filter with the associated mathematical formulation.
+
+    Attributes:
+    GAUSSIAN: Represents a Gaussian filter.
+    BESSEL: Represents a Bessel filter.
+    BUTTERWORTH: Represents a Butterworth filter.
+    LEGENDRE: Represents a Legendre filter.
+    CHEBYSHEV_I: Represents a Chevyshev type I filter.
+    CHEBYSHEV_II: Represents a Chevyshev type II filter.
+    HOURGLASS: Represents a Hourglass filter.
+    ELLIPTIC: Represents a elliptic filter.
+    Custom, rasied-cos, matched, and delay filter types are not available in this release.
+    """
+
+    GAUSSIAN = 0
+    BESSEL = 1
+    BUTTERWORTH = 2
+    LEGENDRE = 3
+    CHEBYSHEV_I = 4
+    CHEBYSHEV_II = 5
+    HOURGLASS = 6
+    ELLIPTIC = 7
+    DELAY = 8
+
+
+#   CUSTOM = 8
+#   RAISED_COS = 9
+#   MATCHED = 10
+#   DELAY = 11
+
+
+class FilterClass(Enum):
+    """Selects class of filter for single band and multiple bands filters.
+
+    Attributes:
+    LOW_PASS: Represents a low pass filter.
+    HIGH_PASS: Represents a high pass filter.
+    DIPLEXER_1: Represents a first group of diplexer filter.
+    BAND_PASS: Represents a band pass filter.
+    BAND_STOP: Represents a band stop filter.
+    DIPLEXER_2: Represents a second group of diplexer filter.
+    LOW_BAND: Represents a combined low pass and multi band filter.
+    BAND_HIGH: Represents a combined high pass and multi band filter.
+    BAND_BAND: Represents a multi band passfilter.
+    STOP_STOP: Represents a multi band stop filter.
+    """
+
+    LOW_PASS = 0
+    HIGH_PASS = 1
+    DIPLEXER_1 = 2
+    BAND_PASS = 3
+    BAND_STOP = 4
+    DIPLEXER_2 = 5
+    LOW_BAND = 6
+    BAND_HIGH = 7
+    BAND_BAND = 8
+    STOP_STOP = 9
+
+
+class FilterImplementation(Enum):
+    """Selects implementations type of filter. This release covers only lumped filter technology.
+
+    Attributes:
+    LUMPED: Represents a lumped implementation.
+    DISTRIB: Represents a distributed implementation.
+    ACTIVE: Represents a active implementation.
+    SWCAP: Represents a switched capacitor implementation.
+    DIGITAL: Represents a digital implementation.
+    """
+
+    LUMPED = 0
+    DISTRIB = 1
+    ACTIVE = 2
+    SWCAP = 3
+    DIGITAL = 4
+
+
+class DiplexerType(Enum):
+    """Selects topology of diplexers.
+
+    Attributes:
+    HI_LO: Represents a high pass low pass diplexer type.
+    BP_1: Represents a band pass band pass diplexer type.
+    BP_2: Represents a band pass band pass diplexer type.
+    BP_BS: Represents a band pass band stop diplexer type.
+    TRIPLEXER_1: Represents a low pass, band pass, and high pass triplexer type.
+    TRIPLEXER_2: Represents a low pass, band pass, and high pass triplexer type.
+    """
+
+    HI_LO = 0
+    BP_1 = 1
+    BP_2 = 2
+    BP_BS = 3
+    TRIPLEXER_1 = 4
+    TRIPLEXER_2 = 5
+
+
+class BesselRipplePercentage(Enum):
+    """Selects peak to peak group delay ripple magnitude as percent of average for Bessl filters.
+    Attributes:
```

Comment:
```suggestion
    """Provides an enum of peak-to-peak group delay ripple magnitudes as percents of averages for Bessl filters.
    
    Attributes:
```
Is "Bessl" correct. Please verify change to description as I was trying to make it more understandable and might have failed.